method.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return true unless block_given?
+
+    (0...length).each do |i|
+      yield(self[i])
+    end
+  end
+
+  def my_each_with_index
+    return true unless block_given?
+
+    k = 0
+    (0...length).each do |i|
+      yield(self[i], k)
+      k += 1
+    end
+  end
+
+  def my_select(&block)
+    return true unless block_given?
+
+    result = []
+    each do |element|
+      result << element if block.call(element) == true
+    end
+    result
+  end
+
+  def my_all?(pattern = nil)
+    if block_given?
+      my_each { |v| return false unless yield(v) }
+    elsif !pattern.nil?
+      my_each { |v| return false unless pattern == v.to_i }
+    else
+      my_each { |v| return false unless v }
+    end
+    true
+  end
+
+  def my_any?(pattern = nil)
+    if block_given?
+      my_each { |v| return true if yield(v) }
+    elsif !pattern.nil?
+      my_each { |v| return true if pattern == v.to_i }
+    else
+      my_each { |v| return true if v }
+      return false
+    end
+    false
+  end
+
+  def my_none?(pattern = nil)
+    if block_given?
+      my_each { |v| return false if yield(v) }
+    elsif !pattern.nil?
+      my_each { |v| return false if pattern == v.to_i }
+    else
+      my_each { |v| return false if v }
+    end
+    true
+  end
+
+  def my_count(num = nil)
+    count = 0
+    if block_given?
+      my_each { |el| count += 1 if yield el }
+    elsif num
+      my_each { |el| count += 1 if el == num }
+    else
+      count = length
+    end
+    count
+  end
+
+  def my_map(&block)
+    return true unless block_given?
+
+    result = []
+    my_each do |i|
+      result << block.call(i)
+    end
+    result
+  end
```

Comment:
1. my_map returns an Enumerator if no block is given

`[1,2,3,4].my_map` should return `#<Enumerator: [1, 2, 3, 4]:each>`

this can be with `return to_enum`